### ISSUE TITLE
getQueryAndParameters on PostgresSQL driver increments parameter values incorrectly

### ISSUE THREAD

**Issue by bmax** [typeorm/typeorm#5514]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
Hello,
```javascript
  const userRepository = connection.getRepository(User);
                const one_column = "new_update";
                const two_column = "hello";
                const qb = userRepository
                    .createQueryBuilder()
                    .update("layout_section_tables")
                    .set({
                        one_column
                    })
                    .where("two_column = :two_column", {
                        two_column
                    });
                console.log(qb.getSql());
                console.log(qb.getSql());
```
The console log returns the following:
```sql
UPDATE "layout_section_tables" SET "one_column" = $1 WHERE "two_column" = $2
UPDATE "layout_section_tables" SET "one_column" = $2 WHERE "two_column" = $2
```

I have identified the problem but not sure if my fix is correct.

Here is where the problem lies: https://github.com/typeorm/typeorm/blob/master/src/query-builder/UpdateQueryBuilder.ts#L390

My solution is adding this.connection.driver instanceof PostgresDriver ||
to the ternary.

I can create the PR if you think that is the best way to go.

Thanks
cc @Kononnable @pleerock

**Comment by imnotjames** [comment]
Not sure if it's the best way but it can't hurt to open the PR and see what tests work and don't!

**Comment by bmax** [comment]
@imnotjames Thanks for the response! We've actually switched to Sequelize for most things but if I can find time I'd love to contribute!

**Comment by imnotjames** [comment]
This should be fixed with #7821
